packages/mobile/src/identity/contactMapping.ts

```diff
@@ -248,15 +248,15 @@ function* fetchWalletAddresses(e164Number: string) {
     // once I've built from the monorepo
     if (!new BigNumber(normalizeAddress(walletAddress)).isZero()) {
       walletToAccountAddress[walletAddress] = accountAddress
-      possibleUserAddresses.push(walletAddress)
+      possibleUserAddresses.add(walletAddress)
     } else {
       // NOTE: Only need this else block if we are not confident all wallets are registered
       walletToAccountAddress[accountAddress] = accountAddress
-      possibleUserAddresses.push(accountAddress)
+      possibleUserAddresses.add(accountAddress)
     }
   }
   yield put(updateWalletToAccountAddress(walletToAccountAddress))
-  return possibleUserAddresses
+  return [...possibleUserAddresses]
```

Comment:
I would just add a small comment here explaining that this is to convert the set into an array. It looks a bit weird otherwise and someone might accidentally remove it.

Reply:
Another option I think is to do `Array.from(possibleUserAddresses)` which wouldn't need the comment imo.

Reply:
Good point, will do that